File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.py
class Solution(object):
    def uncommonFromSentences(self, s1, s2):
        """
        :type s1: str
        :type s2: str
        :rtype: List[str]
        """
        list1 = s1.split(" ")
        list2 = s2.split(" ")
        set1 = {}
        set2 = {}
        result = []
        for l in list1:
            if l not in set1:
                set1[l] = 1
            else:
                set1[l] += 1
        for l in list2:
            if l not in set2:
                set2[l] = 1
            else:
                set2[l] += 1
        for s in set1:
            if s not in set2 and set1[s] == 1:
                result.append(s)
        for s in set2:
            if s not in set1 and set2[s] == 1:
                result.append(s)
        return result
        

File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
        String[] splitter1 = s1.split(" ");
        String[] splitter2 = s2.split(" ");
        Map<String, Integer> mapper = new HashMap<>();
        List<String> answer = new ArrayList<>();
        for(String s: splitter1){
                if(!mapper.containsKey(s)){
                    mapper.put(s, 0);
                }
                mapper.put(s, mapper.get(s) + 1);
        }
        for(String s: splitter2){
                if(!mapper.containsKey(s)){
                    mapper.put(s, 0);
                }
                mapper.put(s, mapper.get(s) + 1);
        }
        
        for(String s: mapper.keySet()){
            if(mapper.get(s) == 1){
                answer.add(s);
            }
        }
        
        return answer.toArray(new String[answer.size()]);
        
    }
}